src/Iocoin/TransactionSigner.cpp

```diff
@@ -0,0 +1,218 @@
+    } else {
+        // Error: Invalid output script
+        return Result<std::vector<Data>>::failure("Invalid output script.");
+    }
+}
+
+template <typename Transaction>
+Data TransactionSigner<Transaction>::createSignature(const Transaction& transaction,
+                                                     const Bitcoin::Script& script, const Data& key,
+                                                     size_t index, Bitcoin::Amount amount
+                                                     ) {
+    auto sighash = transaction.getSignatureHash(script, index, input.hash_type()
+                                                );
+    auto pk = PrivateKey(key);
+    auto sig = pk.signAsDER(Data(begin(sighash), end(sighash)), TWCurveSECP256k1);
+    if (sig.empty()) {
+        return {};
+    }
+    sig.push_back(static_cast<uint8_t>(input.hash_type()));
+    return sig;
+}
+
+template <typename Transaction>
+Data TransactionSigner<Transaction>::pushAll(const std::vector<Data>& results) {
+    auto data = Data{};
+    for (auto& result : results) {
+        if (result.empty()) {
+            data.push_back(OP_0);
+        } else if (result.size() == 1 && result[0] >= 1 && result[0] <= 16) {
+            data.push_back(Bitcoin::Script::encodeNumber(result[0]));
+        } else if (result.size() < OP_PUSHDATA1) {
+            data.push_back(static_cast<uint8_t>(result.size()));
+        } else if (result.size() <= 0xff) {
+            data.push_back(OP_PUSHDATA1);
+            data.push_back(static_cast<uint8_t>(result.size()));
+        } else if (result.size() <= 0xffff) {
+            data.push_back(OP_PUSHDATA2);
+            encode16LE(static_cast<uint16_t>(result.size()), data);
+        } else {
+            data.push_back(OP_PUSHDATA4);
+            encode32LE(static_cast<uint32_t>(result.size()), data);
+        }
+        std::copy(begin(result), end(result), back_inserter(data));
+    }
+    return data;
+}
+
+template <typename Transaction>
+Data TransactionSigner<Transaction>::keyForPublicKeyHash(const Data& hash) const {
+    for (auto& key : input.private_key()) {
+        auto publicKey = PrivateKey(key).getPublicKey(TWPublicKeyTypeSECP256k1);
+        auto keyHash = TW::Hash::ripemd(TW::Hash::sha256(publicKey.bytes));
+        if (std::equal(std::begin(keyHash), std::end(keyHash), std::begin(hash), std::end(hash))) {
+            return Data(key.begin(), key.end());
+        }
+    }
+    return {};
+}
+
+template <typename Transaction>
+Data TransactionSigner<Transaction>::scriptForScriptHash(const Data& hash) const {
+    auto hashString = hex(hash.begin(), hash.end());
+    auto it = input.scripts().find(hashString);
+    if (it == input.scripts().end()) {
+        // Error: Missing redeem script
+        return {};
+    }
+    return Data(it->second.begin(), it->second.end());
+}
+
+// Explicitly instantiate a Signers for compatible transactions.
+template class TW::Iocoin::TransactionSigner<Iocoin::Transaction>;
```

Comment:
I don't think you need to use `template class` and `template <typename Transaction>`